pallets/subtensor/src/swap.rs

```diff
@@ -0,0 +1,422 @@
+        Self::swap_uids(old_hotkey, new_hotkey, &netuid_is_member, &mut weight)?;
+        Self::swap_prometheus(old_hotkey, new_hotkey, &netuid_is_member, &mut weight)?;
+
+        Self::swap_total_hotkey_coldkey_stakes_this_interval(old_hotkey, new_hotkey, &mut weight)?;
+
+        Self::set_last_tx_block(&coldkey, block);
+        weight.saturating_accrue(T::DbWeight::get().writes(1));
+
+        Self::deposit_event(Event::HotkeySwapped {
+            coldkey,
+            old_hotkey: old_hotkey.clone(),
+            new_hotkey: new_hotkey.clone(),
+        });
+
+        Ok(Some(weight).into())
+    }
+
+    /// Retrieves the network membership status for a given hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The hotkey to check for network membership.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<Vec<u16>, Error<T>>` - A vector of network IDs where the hotkey is a member.
+    pub fn get_netuid_is_member(
+        old_hotkey: &T::AccountId,
+        weight: &mut Weight,
+    ) -> Result<Vec<u16>, Error<T>> {
+        let netuid_is_member: Vec<u16> =
+            <IsNetworkMember<T> as IterableStorageDoubleMap<_, _, _>>::iter_prefix(old_hotkey)
+                .map(|(netuid, _)| netuid)
+                .collect();
+        weight.saturating_accrue(T::DbWeight::get().reads(netuid_is_member.len() as u64));
+        Ok(netuid_is_member)
+    }
+
+    /// Swaps the owner of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `coldkey` - The coldkey owning the hotkey.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_owner(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        coldkey: &T::AccountId,
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        Owner::<T>::remove(old_hotkey);
+        Owner::<T>::insert(new_hotkey, coldkey.clone());
+        weight.saturating_accrue(T::DbWeight::get().writes(2));
+        Ok(())
+    }
+
+    /// Swaps the total stake of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_total_hotkey_stake(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        if let Ok(total_hotkey_stake) = TotalHotkeyStake::<T>::try_get(old_hotkey) {
+            TotalHotkeyStake::<T>::remove(old_hotkey);
+            TotalHotkeyStake::<T>::insert(new_hotkey, total_hotkey_stake);
+            weight.saturating_accrue(T::DbWeight::get().writes(2));
+        }
+        Ok(())
```

Comment:
This is a noop if the old_hotkey doesn't exist, so unless that's sometimes expected, this should prob return an error in that case

Can also add some debug logs

Reply:
If the hotkey doesnt exist , it would be caught here : https://github.com/opentensor/subtensor/pull/542/files#diff-f17d3c10f0bee92182f2bdb18feb020e8e740d95394451f8c8930a219e1c0b7aR43 . Do you think its sufficient , or do we still need to catch the error in this function?

Reply:
Additionally, the weight calculation here is missing the read from `try_get`.